src/commands/farm.rs

```diff
@@ -135,31 +136,60 @@ async fn start_farming(farming_args: FarmingArgs) -> Result<(Farmer, Node)> {
 
 #[instrument]
 async fn prepare_farming() -> Result<FarmingArgs> {
-    let config_args = parse_config()?;
-
-    let node_name = config_args.node_config_args.name;
-    let chain = match config_args.node_config_args.chain.as_str() {
+    let config = parse_config()?;
+
+    let node_config = config.node;
+    let NodeConfig {
+        chain,
+        execution: _,
+        blocks_pruning,
+        state_pruning,
+        validator,
+        name,
+        port,
+        unsafe_ws_external,
+    } = node_config;
+
+    let chain = match chain.as_str() {
         "dev" => chain_spec::dev_config().unwrap(),
         _ => unreachable!("there are no other valid chain-specs at the moment"),
     };
-    let is_validator = config_args.node_config_args.validator;
-    let role = match is_validator {
+    let role = match validator {
         true => subspace_sdk::node::Role::Authority,
         false => subspace_sdk::node::Role::Full,
     };
+    let rpc_method = match unsafe_ws_external {
+        true => RpcMethods::Unsafe,
+        false => RpcMethods::Auto,
+    };
```

Comment:
Bools are kinda undescriptive. Maybe we could just use enums which will speak for themselves in config. Like:
```toml
rpc_method = "auto"
role = "authority"
```

Reply:
we wanted to be config file to be in its most simplest form for the user. Turning them into enums, would be better for me (and the developers), but I would argue against it for the users. Matching against a boolean is tolerable I guess for the sake of simplifying the user experience.

Reply:
Ok I think I found a good way that will satisfy both of us :)